flake.nix

```diff
@@ -62,7 +62,7 @@
           mkdir -p $out/include
           cp ${src}/llama.h $out/include/
         '';
-        cmakeFlags = [ "-DLLAMA_BUILD_SERVER=ON" "-DLLAMA_MPI=ON" "-DBUILD_SHARED_LIBS=ON" "-DCMAKE_SKIP_BUILD_RPATH=ON" ];
+        cmakeFlags = [ "-DLLAMA_NATIVE=OFF" "-DLLAMA_BUILD_SERVER=ON" "-DLLAMA_MPI=ON" "-DBUILD_SHARED_LIBS=ON" "-DCMAKE_SKIP_BUILD_RPATH=ON" ];
```

Comment:
And enable `LLAMA_AVX` `LLAMA_AVX2` `LLAMA_FMA` `LLAMA_F16C` , to get the previous default behavoir back.
`-arch=native` does nothing when using nix, bc for nix, the same input,should always produce the same output, independent of host machine. :)

Reply:
while you are at it, you can disable MPI, since that was broken in a recently merged pr.

Reply:
Actually with `DLLAMA_NATIVE=OFF` you already get `-mf16c -mfma -mavx -mavx2` by default which matches the previous behavior. You have to turn on `LLAMA_NATIVE` or not set it in order for `march=native` to be used.
I've removed the MPI flag in the latest commit (defaults to off).